xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-ui/src/main/resources/XWiki/DeletedDocuments.xml

```diff
@@ -341,6 +342,46 @@ XWiki.index.trash.documents.confirmDelete = function(event, table, rowIdx) {
       );
   }
 }
+
+/**
+ * Event listener, called when clicking the "delete all" button.
+ *
+ * @param event the link activation event
+ */
+XWiki.index.trash.documents.confirmDeleteAll = function(event) {
+  event.stop();
+  if (this.disabled) {
```

Comment:
Shouldn't this be updated to use the CSS class instead?

Reply:
Not sure it's necessary: this check is related to the `this.disabled = true` in the `onCreate`, so it still works. Apparently bootstrap deactivate the link when the `disabled` class is used, so we did not event enter this function.

Reply:
Yes, but the delete button should appear as disabled while the delete operation is in progress. I'm talking about this flow:

* the button is enabled
* the user clicks on it
* this method is called
* you make an AJAX request
* the button should visually appear as disabled until we receive a response

Reply:
Indeed. I'll check